snapcraft/internal/pluginhandler/_mangling.py

```diff
@@ -0,0 +1,83 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import logging
+import os
+import re
+from typing import FrozenSet, List
+
+from snapcraft import file_utils
+from snapcraft.internal import elf, repo
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_dynamic_linker(library_list: List[str]) -> str:
+    """Return the dynamic linker from library_list."""
+    regex = re.compile(r'(?P<dynamic_linker>ld-[\d.]+.so)$')
+
+    for library in library_list:
+        m = regex.search(os.path.basename(library))
+        if m:
+            return m.group('dynamic_linker')
+
+    raise NotImplementedError(
+        'The format for the linker should be of the form '
+        '<root>/ld-<X>.<Y>.so. There are no matches for the '
+        'current libc6 package')
+
+
+def handle_glibc_mismatch(*, elf_files: FrozenSet[elf.ElfFile],
+                          root_path: str, core_base_path: str,
+                          snap_base_path: str) -> None:
+    formatted_list = list()  # type: List[str]
+    patch_elf_files = list()  # type: List[elf.ElfFile]
+    for elf_file in elf_files:
+        required_glibc = elf_file.get_required_glibc()
+        if required_glibc:
+            formatted_list.append('- {} -> GLIBC {}'.format(
+                elf_file.path, required_glibc))
+            patch_elf_files.append(elf_file)
+    logger.warning('The primed files will not work with the current '
+                   'base given the GLIBC mismatch of the primed '
+                   'files and the linker version (2.23) used in the '
+                   'base. These are the GLIBC versions required by '
+                   'the primed files that do not match and will be '
+                   'patched:\n {}\n'.format(
```

Comment:
This message implies (to me) that we're patching the mismatching files to use the older version of libc, which isn't the case. We're migrating the newer libc into the snap and patching them to use that instead of the one in the core snap, right? Perhaps this warning could say something like:

"The primed files will not work[...] These are the GLIBC versions required by the primed files that don't match[...]. To work around this, the newer libc will be migrated into the snap, and these files will be patched to use it."